src/test/java/io/jenkins/pluginhealth/scoring/probes/ProbeEngineTest.java

```diff
@@ -0,0 +1,97 @@
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.jenkins.pluginhealth.scoring.probes;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.atMostOnce;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.time.ZonedDateTime;
+import java.util.List;
+import java.util.stream.Stream;
+
+import io.jenkins.pluginhealth.scoring.model.Plugin;
+import io.jenkins.pluginhealth.scoring.model.ProbeResult;
+import io.jenkins.pluginhealth.scoring.service.PluginService;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class ProbeEngineTest {
+
+    @Mock
+    private PluginService pluginService;
+
+    @Test
+    public void shouldBeAbleToRunSimpleProbe() {
+        final Plugin plugin = mock(Plugin.class);
+        final Probe probe = mock(Probe.class);
+        final ProbeEngine probeEngine = new ProbeEngine(List.of(probe), pluginService);
+        final ProbeResult expectedResult = ProbeResult.success("foo", "bar");
+
+        when(probe.doApply(plugin)).thenReturn(expectedResult);
+        when(pluginService.streamAll()).thenReturn(Stream.of(plugin));
+        probeEngine.run();
+
+        verify(probe).doApply(plugin);
+        verify(plugin, atMostOnce()).addDetails(expectedResult);
+        verify(pluginService, atMostOnce()).saveOrUpdate(plugin);
+    }
+
+    @Test
+    public void shouldNotAccessPluginWithPastResultAndReleaseRequirement() {
+        final Plugin plugin = new Plugin("foo", "bar", ZonedDateTime.now().minusDays(1))
+            .addDetails(ProbeResult.success("wiz", "This is good"));
+        final Probe probe = mock(Probe.class);
+        final ProbeEngine probeEngine = new ProbeEngine(List.of(probe), pluginService);
+
+        when(probe.requiresRelease()).thenReturn(Boolean.TRUE);
+        when(probe.key()).thenReturn("wiz");
+        when(pluginService.streamAll()).thenReturn(Stream.of(plugin));
+        probeEngine.run();
+
+        verify(probe, never()).doApply(plugin);
+    }
+
+    @Test
+    public void shouldNotSaveErrors() throws Exception {
+        final Plugin plugin = mock(Plugin.class);
+        final Probe probe = mock(Probe.class);
+        final ProbeEngine probeEngine = new ProbeEngine(List.of(probe), pluginService);
+
+        when(probe.doApply(any(Plugin.class))).thenReturn(ProbeResult.error("foo", "bar"));
+        when(pluginService.streamAll()).thenReturn(Stream.of(plugin));
+        probeEngine.run();
+
+        verify(plugin, never()).addDetails(any());
```

Comment:
This test is to make sure that when our probe results in an error, we don't save it inside the plugin's `details` map (by calling the `addDetails` method).

Because you've filtered out the `ResultStatus.ERROR` status from reaching to `addDetails` method.

Reply:
exactly. I still don't know if it's a good idea or not. But for the moment I keep that. If I change my mind, I'll change this test as well.

This good because if we have some code where we rely on this behavior and we change it, we will see the problem at build time (on CI) and not when we run the code (production).